sky/packages/sky/lib/src/rendering/paragraph.dart

```diff
@@ -59,22 +59,27 @@ class RenderParagraph extends RenderBox {
     textPainter.minHeight = constraints.minHeight;
     textPainter.maxHeight = constraints.maxHeight;
     textPainter.layout();
+    // By default, we shrinkwrap to the intrinsic width.
+    double width = constraints.constrainWidth(textPainter.maxIntrinsicWidth);
```

Comment:
LGTM, but I'm still trying to understand the exact issue.

What width were we using before that was causing the text to be off-screen when right-aligned?